tests/test_page.py

```diff
@@ -395,3 +396,142 @@ async def test_expose_function_should_work_with_complex_objects(page, server):
+    for [value, message] in cases:
+        [error, _] = await asyncio.gather(
+            page.waitForEvent("pageerror"),
+            page.evaluate(f"() => setTimeout(() => {{ throw {value}; }}, 0)"),
+        )
+        assert (
+            error.message == ("uncaught exception: " + message) if is_firefox else value
+        )
+
+
+@pytest.mark.skip_browser("firefox")
+async def test_page_error_should_handle_object(page, is_chromium):
+    # Firefox just does not report this error.
+    [error, _] = await asyncio.gather(
+        page.waitForEvent("pageerror"),
+        page.evaluate("() => setTimeout(() => { throw {}; }, 0)"),
+    )
+    assert error.message == "Object" if is_chromium else "[object Object]"
+
+
+@pytest.mark.skip_browser("firefox")
+async def test_page_error_should_handle_window(page, is_chromium):
+    # Firefox just does not report this error.
+    [error, _] = await asyncio.gather(
+        page.waitForEvent("pageerror"),
+        page.evaluate("() => setTimeout(() => { throw window; }, 0)"),
+    )
+    assert error.message == "Window" if is_chromium else "[object Window]"
+
+
+expected_output = "<html><head></head><body><div>hello</div></body></html>"
+
+
+async def test_set_content_should_work(page, server):
+    await page.setContent("<div>hello</div>")
+    result = await page.content()
+    assert result == expected_output
+
+
+async def test_set_content_should_work_with_domcontentloaded(page, server):
+    await page.setContent("<div>hello</div>", waitUntil="domcontentloaded")
+    result = await page.content()
+    assert result == expected_output
+
+
+async def test_set_content_should_work_with_doctype(page, server):
+    doctype = "<!DOCTYPE html>"
+    await page.setContent(f"{doctype}<div>hello</div>")
+    result = await page.content()
+    assert result == f"{doctype}{expected_output}"
+
+
+async def test_set_content_should_work_with_HTML_4_doctype(page, server):
+    doctype = '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">'
+    await page.setContent(f"{doctype}<div>hello</div>")
+    result = await page.content()
+    assert result == f"{doctype}{expected_output}"
+
+
+async def test_set_content_should_respect_timeout(page, server):
+    img_path = "/img.png"
+    # stall for image
+    server.set_route(img_path, lambda request: None)
+    error = None
+    try:
+        await page.setContent(
+            '<img src="${server.PREFIX + img_path}"></img>', timeout=1
+        )
+    except Error as e:
+        error = e
+    assert isinstance(error, TimeoutError)
+
+
+async def test_set_content_should_respect_default_navigation_timeout(page, server):
+    page.setDefaultNavigationTimeout(1)
+    img_path = "/img.png"
+    # stall for image
+    await page.route(img_path, lambda route, request: None)
+
+    error = None
+    try:
+        await page.setContent(f'<img src="{server.PREFIX + img_path}"></img>')
+    except Error as e:
```

Comment:
Nit, would maybe simply it in the future: 
```py
import pytest

def test_zero_division():
    with pytest.raises(ZeroDivisionError):
        1 / 0
```

Reply:
Will it le me test exception properties?

Reply:
Unfortunately not. But good to just check if an exception was thrown which we do in some tests.